File: code-ninja-course/JAVA/src/main/java/com/codeninja/words/Words.java
package com.codeninja.words;
import java.io.IOException;

	public class Words {
	public int find() throws IOException
	{
	    System.out.println("Enter a list of words followed by end-of-file: ");

	    int vector_sz = 0;
	    java.util.Vector<String> words = new java.util.Vector<String>();
	    String word;
  
	    java.io.BufferedReader reader = new java.io.BufferedReader(new java.io.InputStreamReader( System.in));
	    while ((word = reader.readLine()) != null)
	    {
	        words.add(word);
	    }

	    vector_sz = words.size();
	    if(vector_sz == 0)
	    {
	        System.out.println("Please enter some words!");
	        return -1;
	    }

	    java.util.Vector<Integer> frequency = new java.util.Vector<>();
	    java.util.Vector<String> unique_words = new java.util.Vector<String>();
	    int count = 0;
	    boolean isUnique;

	    unique_words.add(words.get(0));

	    for(Integer i = 0; i < words.size(); ++i)
	    {
	        isUnique = true;

	        for(int j = 0; j < unique_words.size(); ++j)
	        {
	            if(words.get(i) == unique_words.get(j))
	            {
	                isUnique = false;
	            }
	        }
	      if(isUnique)
	      {
	          unique_words.add(words.get(i));
	      }
	    }

	    for(Integer i = 0; i < unique_words.size(); ++i)
	    {
	        for(int j = 0; j < words.size(); ++j)
	        {
	            if(unique_words.elementAt(i) == words.elementAt(j))
	            {
	                count += 1;
	            }
	        }
	        frequency.add(count);
	        count = 0;
	    }

	    for(int i = 0; i < unique_words.size(); ++i)
	    {
	        System.out.printf(unique_words.elementAt(i) + "   %d /n", frequency.get(i) );
	    }
	    return 0;

}
}

File: code-ninja-course/CSHARP/Code/Words/Words.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CleanCode
{
    public class Words
    {
        public int find()
        {
            Console.WriteLine("Enter a list of words followed by end-of-file: ");

            int list_sz = 0;
            List<string> words = new List<string>();
            string word;

            while ((word = Console.ReadLine()) != null)
            {
                words.Add(word);
            }

            list_sz = words.Count;

            if (list_sz == 0)
            {
                Console.WriteLine("Please enter some words!");
                return -1;
            }

            System.Collections.Generic.List<int> frequency = new System.Collections.Generic.List<int>();
            System.Collections.Generic.List<string> unique_words = new System.Collections.Generic.List<String>();

            int count = 0;
            bool isUnique;

            unique_words.Add(words[0]);

            for (Int32 i = 0; i < words.Count; ++i)
            {
                isUnique = true;

                for (int j = 0; j < unique_words.Count; ++j)
                {
                    if (words[i] == unique_words[j])
                    {
                        isUnique = false;
                    }
                }
                if (isUnique)
                {
                    unique_words.Add(words[i]);
                }
            }

            for (Int32 i = 0; i < unique_words.Count; ++i)
            {
                for (int j = 0; j < words.Count; ++j)
                {
                    if (unique_words[i] == words[j])
                    {
                        count += 1;
                    }
                }
                frequency.Add(count);
                count = 0;
            }

            for (int i = 0; i < unique_words.Count; ++i)
            {
                Console.WriteLine(unique_words[i] + "   %d /n", frequency[i]);
            }
            return 0;
        }
    }
}
